cloud_build/verify_provenance.sh

```diff
@@ -14,8 +14,11 @@ echo "Installing jq using apt..."
 apt update && apt install jq=1.5+dfsg-1.3 -y
 
 # Download slsa-verifier in order to validate the docker image provenance.
+# This takes the version of slsa-verifier defined in tooling/go.mod.
 echo "Installing slsa-verifier using go..."
-go install github.com/slsa-framework/slsa-verifier/v2/cli/slsa-verifier@a43888265e1f6aae98c924538298944f2721dcf0 #v2.0.1: https://github.com/slsa-framework/slsa-verifier/releases/tag/v2.0.1
+cd tooling
+grep _ tooling_test.go | cut -f2 -d '"' | xargs -n1 -t go install
```

Comment:
I updated the script to be a bit more clear. We are installing based on the slsa-verifier hash in go.sum (for example, [this is the current hash it will install)](https://github.com/flutter/cocoon/pull/2546/files#diff-d7d4506d5dd0bc83864e72196eebca4c1e780e82ef5a68589f8cbd7d69324841R765), but the benefit is that now Dependabot will update this hash and version automatically